src/main/java/com/yahoo/bullet/bql/tree/StreamNode.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright 2018, Oath Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.bql.tree;
+
+import lombok.Getter;
+
+import java.util.Objects;
+
+@Getter
+public class StreamNode extends Node {
+    private final String timeDuration;
+
+    public StreamNode(String timeDuration, NodeLocation nodeLocation) {
+        super(nodeLocation);
+        this.timeDuration = timeDuration;
+    }
+
+    @Override
+    public <R, C> R accept(ASTVisitor<R, C> visitor, C context) {
+        return visitor.visitStream(this, context);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
```

Comment:
We can probably push these two if statements to somewhere common since we absolutely use them a ridiculous number of times

Reply:
might just stick @equalsandhashcode on everything then

Reply:
Might be overkill and will add a lot of codepaths that stay uncovered since lombok adds a canEqual and whatnot that can't normally be reached. Also hashcode seems some custom hash function with primes. Alright just forget about this.